FreeRTOS/FreeRTOS-Kernel
# Adding GIC v2 interrupt group handling to GCC_AARCH64 port.

Description
-----------

re: https://github.com/FreeRTOS/FreeRTOS-Kernel/issues/316#issuecomment-2327254371    
Adding handling for GIC V2 interrupt grouping and security extension. Additional handling determines which interrupt group is asserted by inspecting HPPIR. Based on the interrupt group IAR/EOIR or AIAR/AEOIR registers are used to acknowledge and end interrupts.

Test Steps
-----------

Test target used = STM32MP357F-EV1.  
  
Manually verified interrupt acknowledgement, routing, and ending for group 0 and group 1 interrupts by utilizing the ARMV8-A physical timer interrupt and varying the security routing of the interrupt.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
